lib/surface/api.ex

```diff
@@ -100,9 +101,34 @@ defmodule Surface.API do
   end
 
   @doc false
-  def put_assign!(caller, func, name, type, opts, opts_ast, line) do
```

Comment:
would it make sense to keep this signature around for now, deprecate it, and remove it in `0.6`?